src/compiler/binder.ts

```diff
@@ -533,6 +534,7 @@ namespace ts {
                 }
             }
 
+            symbol.isDeprecated = symbol.isDeprecated || isDeprecated;

```

Comment:
SHould this be SymbolFlags instead?

Reply:
Alternatively, making this a ModifierFlag would go well with public/@public et al. It would also go better with `getNodeModifiers` in services.